### PR TITLE
WIP: Sgx build improvements

### PR DESCRIPTION
**Pull request by dreemkiller**
Some minor changes to the SGX `build.rs` and Makefiles to reduce redundancy in builds and make sure changes get picked up

### CODE REVIEW

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 31a16f14f3842d2418762b9a6e9fff52913eee3d
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=65f4c3e0-12f9-40b5-863f-49fd65fd9c32%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 59b5f8cd059ba0eb2d96a780516686094882238a
* Result: SUCCEEDED
* [Build Logs](https://01evtei465.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=bf7500ed-0e18-4a63-afbc-faab1d7b0c1c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: sgx_buildspec.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+version: 0.2
+
+env:
+  shell: bash
+
+phases:
+  pre_build:
+    commands:
+      # Clean the repostory, which automatically downloads the rust toolchain specified in the repostory
```
Comment (dominic-mulligan-arm) [inline sgx_buildspec.yml:9]: sp: repostory --> repository

File: trustzone_buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+env:
+  shell: bash
+
+phases:
+  pre_build:
+    commands:
+      # Set up the environment for trustzone backend
+      - pushd /work/rust-optee-trustzone-sdk && source environment && popd
+      # Clean the repostory, which automatically downloads the rust toolchain specified in the repostory
```
Comment (dominic-mulligan-arm) [inline trustzone_buildspec.yml:11]: sp: repostory --> repository

**Comment by dominic-mulligan-arm** [comment]
@dreemkiller what is the relationship between this changeset and #195?  Does anything in #195 need porting into this changeset (given @geky is away, and both deal with similar topics)?  If not, then I'll approve this.

**Comment by dreemkiller** [comment]
There is coincidental overlap between #195 and this one.

As the author of #195, I would prefer we use #195 and eventually merge that, but I am not an impartial party :-)

**Comment by dominic-mulligan-arm** [comment]
This is still marked WIP in the title, is it in a state ready to merge (eyeing the backlog of PRs and trying to work out which order is best to merge)?

**Comment by dreemkiller** [comment]
This one can be low-priority to merge. I need to look at @geky 's PR and see if I can use it to improve this one (or not).

Imma keep it WIP for a bit.

**Comment by dreemkiller** [comment]
A good portion of these have been done in other PRs. Closing.

**Pull request closed without merging**